### PR TITLE
docs: add opa example

### PR DESCRIPTION
**Pull request by hugorut**
* adds opa example support

### CODE REVIEW

**Comment by hugorut** [comment]
conditional output tested here: https://dev.azure.com/infracost/base/_build/results?buildId=550&view=logs&j=d0ee13d9-3503-51dd-3607-d7bccb219083&t=a4365896-62ee-59cd-f8d1-dcde0547daba

File: .azure/pipelines/examples-test.yml [hunk 1]
```diff
@@ -130,6 +130,51 @@ jobs:
...
+      - bash: >-
+          opa eval --input /tmp/infracost.json -d
+          examples/opa/policy/policy.rego --format pretty "data.infracost.deny"
+          | tee /tmp/opa.out
+        displayName: Run OPA
+      - bash: |
+          denyReasons=$(</tmp/opa.out)
+          if [ "$denyReasons" != "[]" ]; then
+            echo "##vso[task.logissue type=error]Policy check failed"
+            echo "##vso[task.logissue type=error]$denyReasons"
```
Comment (vdmgolub) [inline .azure/pipelines/examples-test.yml:172]: The output shows this, do we need to iterate through the list? https://dev.azure.com/infracost/base/_build/results?buildId=550&view=logs&j=d0ee13d9-3503-51dd-3607-d7bccb219083&t=a4365896-62ee-59cd-f8d1-dcde0547daba&l=13

> ##[error]n

File: examples/opa/README.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
+# Open Policy Agent Example
+
+This example shows how to set cost policies using [Open Policy Agent](https://www.openpolicyagent.org/).  For simplicity, this is based off the terraform-plan-json example, which does not require Terraform to be installed.
+
+When the policy checks pass, the Azure Devops pipeline step called "Check Policies" passes and outputs `Policy check passed.` in the task logs. When the policy checks fail, that step fails and the task logs show the details of the failing policies.
```
Comment (vdmgolub) [inline examples/opa/README.md:5]:
```suggestion
When the policy checks pass, the Azure DevOps pipeline step called "Check Policies" passes and outputs `Policy check passed.` in the task logs. When the policy checks fail, that step fails and the task logs show the details of the failing policies.
```

File: .azure/pipelines/examples-test.yml (hunk 1 above)
Comment (hugorut) [reply .azure/pipelines/examples-test.yml:172]: That output was just for testing, the actual output would be something like `"['resource_1', 'resource_2']"`

**Review by vdmgolub — APPROVED** [review]
👍 👍

**Pull request merged**